backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -2251,54 +2261,148 @@ public void deleteLogVersionedInputs(Session session, Long repoId, String commit
+        artifactEntity.setUploadCompleted(false);
+        session.getTransaction().commit();
       }
-      return new AbstractMap.SimpleEntry<>(null, null);
+    } else {
+      uploadId = null;
+    }
+    return new AbstractMap.SimpleEntry<>(artifactEntity.getPath(), uploadId);
+  }
+
+  @Override
+  public Response commitArtifactPart(CommitArtifactPart request) throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      ArtifactEntity artifactEntity = getArtifactEntity(session, request.getId(), request.getKey());
+      ArtifactPart artifactPart = request.getArtifactPart();
+      ArtifactPartEntity artifactPartEntity =
+          new ArtifactPartEntity(
+              artifactEntity, artifactPart.getPartNumber(), artifactPart.getEtag());
+      session.beginTransaction();
+      session.saveOrUpdate(artifactPartEntity);
+      session.getTransaction().commit();
+      return Response.newBuilder().build();
+    }
+  }
+
+  @Override
+  public GetCommittedArtifactParts.Response getCommittedArtifactParts(
+      GetCommittedArtifactParts request) throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Set<ArtifactPartEntity> artifactPartEntities =
+          getArtifactPartEntities(session, request.getId(), request.getKey());
+      GetCommittedArtifactParts.Response.Builder response =
+          GetCommittedArtifactParts.Response.newBuilder();
+      artifactPartEntities.forEach(
+          artifactPartEntity -> response.addArtifactParts(artifactPartEntity.toProto()));
+      return response.build();
+    }
+  }
+
+  private Set<ArtifactPartEntity> getArtifactPartEntities(
+      Session session, String experimentRunId, String key) throws ModelDBException {
+    ArtifactEntity artifactEntity = getArtifactEntity(session, experimentRunId, key);
+    return artifactEntity.getArtifactPartEntities();
+  }
+
+  @Override
+  public CommitMultipartArtifact.Response commitMultipartArtifact(
+      CommitMultipartArtifact request, CommitMultipartFunction commitMultipartFunction)
+      throws ModelDBException {
+    SimpleEntry<String, String> result;
+    List<PartETag> partETags;
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      ArtifactEntity artifactEntity = getArtifactEntity(session, request.getId(), request.getKey());
+      Set<ArtifactPartEntity> artifactPartEntities = artifactEntity.getArtifactPartEntities();
+      partETags =
+          artifactPartEntities.stream()
+              .map(ArtifactPartEntity::toPartETag)
+              .collect(Collectors.toList());
+      session.beginTransaction();
+      artifactEntity.setUploadCompleted(true);
+      artifactPartEntities.forEach(session::delete);
+      artifactPartEntities.clear();
+      session.getTransaction().commit();
+      result = getS3PathAndMultipartUploadId(session, artifactEntity, true, request.getKey(), null);
```

Comment:
`getS3PathAndMultipartUploadId` has `artifactEntity.setUploadCompleted(false);`. Seems like calling it here is a bug (or at least something that will probably be a bug in the future).

Reply:
what I wanted to do with getS3PathAndMultipartUploadId is to just get the information from artifactEntity. But it can be done without getS3PathAndMultipartUploadId. made a refactoring to not use it.